tests/indexes/tests.py

```diff
@@ -452,6 +453,40 @@ def test_is_null_condition(self):
                 ))
             editor.remove_index(index=index, model=Article)
 
+    @skipUnlessDBFeature('supports_expression_indexes')
+    def test_func_index(self):
+        index_name = 'partial_func_idx'
+        index = Index(
+            Lower('headline').desc(),
+            name=index_name,
+            condition=Q(pub_date__isnull=False),
+        )
+        # Add index.
+        with connection.schema_editor() as editor:
+            editor.add_index(index=index, model=Article)
+            sql = index.create_sql(Article, schema_editor=editor)
+            table = Article._meta.db_table
+            constraints = connection.introspection.get_constraints(
+                cursor=connection.cursor(), table_name=table,
+            )
+            self.assertIn(index_name, constraints)
+            self.assertIs(sql.references_column(table, 'headline'), True)
+            # SQLite does not support introspection of expressions in indexes
+            # https://www.sqlite.org/pragma.html#pragma_index_xinfo
```

Comment:
IMO, this is not related with `index_xinfo` but with naive parsing in introspection on SQLite. I added `_get_index_columns_orders` to fix this issue.